GDJS/GDJS/Events/CodeGeneration/MetadataDeclarationHelper.cpp

```diff
@@ -0,0 +1,1467 @@
+        GetExtensionIconUrl(extension), GetExtensionIconUrl(extension));
+
+    DeclareEventsFunctionParameters(eventsBasedObject.GetEventsFunctions(),
+                                    eventsFunction, action, 1);
+    return action;
+  }
+}
+
+gd::String MetadataDeclarationHelper::GetStringifiedExtraInfo(
+    const gd::PropertyDescriptor &property) {
+  gd::String stringifiedExtraInfo = "";
+  if (property.GetType() == "Choice") {
+    stringifiedExtraInfo += "[";
+    for (size_t i = 0; i < property.GetExtraInfo().size(); i++) {
+      stringifiedExtraInfo += property.GetExtraInfo().at(i);
+      if (i < property.GetExtraInfo().size() - 1) {
+        stringifiedExtraInfo += ",";
+      }
+    }
+    stringifiedExtraInfo += "]";
+  }
+  return stringifiedExtraInfo;
+}
+
+gd::String
+MetadataDeclarationHelper::UncapitalizeFirstLetter(const gd::String &string) {
+  return string.size() < 1 ? string
+                           : string.substr(0, 1).LowerCase() + string.substr(1);
+}
+
+gd::String
+MetadataDeclarationHelper::CapitalizeFirstLetter(const gd::String &string) {
+  return string.size() < 1 ? string
+                           : string.substr(0, 1).UpperCase() + string.substr(1);
+}
+
+void MetadataDeclarationHelper::DeclarePropertyInstructionAndExpression(
+    gd::PlatformExtension &extension,
+    gd::InstructionOrExpressionContainerMetadata &entityMetadata,
+    const gd::AbstractEventsBasedEntity &eventsBasedEntity,
+    const gd::NamedPropertyDescriptor &property,
+    const gd::String &propertyLabel, const gd::String &expressionName,
+    const gd::String &conditionName, const gd::String &actionName,
+    const gd::String &toggleActionName, const gd::String &setterName,
+    const gd::String &getterName, const gd::String &toggleFunctionName,
+    const int valueParameterIndex,
+    std::function<gd::AbstractFunctionMetadata &(
+        gd::AbstractFunctionMetadata &instructionOrExpression)>
+        addObjectAndBehaviorParameters) {
+  auto &propertyType = property.GetType();
+
+  auto uncapitalizedLabel =
+      UncapitalizeFirstLetter(property.GetLabel() || property.GetName());
+  if (propertyType == "Boolean") {
+    auto &conditionMetadata = entityMetadata.AddScopedCondition(
+        conditionName, propertyLabel,
+        _("Check the property value for") + " " + uncapitalizedLabel + ".",
+        _("Property") + " " + uncapitalizedLabel + " " + "of _PARAM0_ is true",
+        eventsBasedEntity.GetFullName() || eventsBasedEntity.GetName(),
+        GetExtensionIconUrl(extension), GetExtensionIconUrl(extension));
+    addObjectAndBehaviorParameters(conditionMetadata);
+    conditionMetadata.SetFunctionName(getterName);
+
+    auto &setterActionMetadata = entityMetadata.AddScopedAction(
+        actionName, propertyLabel,
+        _("Update the property value for") + " \"" + uncapitalizedLabel +
+            _("\"."),
+        _("Set property value for") + " " + uncapitalizedLabel + " " +
```

Comment:
Same everywhere, use a replace with a string like `Set property value for <property_name> of _PARAM0_ to <parameter>`